### ISSUE TITLE
`Unnecessary _nowPlaying change` assertion failure

### ISSUE THREAD

**Issue by kunalzigma** [sbooth/SFBAudioEngine#366]
After rigorous testing for our audio requirement which is fast playback of a different sounds multiple times but individually, we see this happen. 

<img width="1402" alt="Screenshot 2024-10-23 at 3 11 23 PM" src="https://github.com/user-attachments/assets/10e222bf-efd4-4b93-a165-b819590e34db">
<img width="1405" alt="Screenshot 2024-10-23 at 3 11 30 PM" src="https://github.com/user-attachments/assets/610dc5eb-6168-429f-8718-77dda89fcbb3">
[crashReport2310.pdf](https://github.com/user-attachments/files/17488602/crashReport2310.pdf)

Was able to recreate this, by playing audio and then check if it isplaying and pausing it in applicationWillResignActive. But now cannot recreate it again. Happens rarely like once a day in 2 hour session a day.

<img width="2192" alt="Screenshot 2024-10-23 at 3 39 33 PM" src="https://github.com/user-attachments/assets/66e62cb1-64ce-4502-8ddb-8dc5b0dad42a">
![Uploading Screenshot 2024-10-23 at 3.39.33 PM.png…]()

**Comment by sbooth** [comment]
I appreciate the report, especially on iOS as I primarily use SFBAudioEngine on macOS.

The potential good news is that this shouldn't cause any major problems in non-debug code since the assert is disabled. That being said, it shouldn't happen.

I'll need to find a way to reliably trigger the issue to debug/fix it. Since the callbacks are asynchronous it might take some work to determine a way to reproduce the problem.

**Comment by kunalzigma** [comment]
Thank you for the library. I have tried almost every AVAudioEngine Library for my requirements but avaudioengine always have crashes in every library. Even tried avaudioengine myself, i get even more crashes. 
Currently using AVAudioPlayer, though sounds are not as per my expectations but it is reliable. No crashes. Only issue with AVAudioplayer is when app is running on efficiency cores, it does not performs as intended whereas on performance cores, it is great. And as a developer, i think i cannot control which core my app is running on. Tried a lot. 
This library does the job for us in efficiency cores.

Thats certainly good news, if this wont crash on user's device. And will changing the play url get back to normal playback in that particular situation? Users wont mind that, i think.
